### ISSUE TITLE
link vimrc and .vim/ in $HOME

### ISSUE THREAD

**Issue by genebean** [nix-community/home-manager#4464]
### Description

It is very confusing to me the configuration file and plugins directory for vim are not linked in $HOME. It seems to me that there should be symlinks in $HOME like there are when configuring bat and zsh.

**Comment by genebean** [comment]
FWIW, when I am trying to wrap my head around what was actually laid down on disk it is helpful to be able to find the file in the expected location. Additionally, if home manager manages a link there then it prevents a local .vimrc from being used instead as that path is read first.

**Comment by berbiche** [comment]
Hi,

I believe the vim/neovim package generated and managed by Home Manager use specific flags to point the vim/neovim binary to the configuration in the nix store.

If you look at the content of the vim "binary", you'll realise that it's a wrapper that forces a specific vimrc to be used: `head -n30 "$(command -v vim)" | cat -v`.

The advantage of this solution is that you can test different vim configurations without overlap.

**Comment by genebean** [comment]
True, but the UX of not having a file, or a symlink to a file, in the place that we’ve all been trained to look for it is not good, and rather confusing.

**Comment by trueNAHO** [comment]
As far as I am aware, several Home Manager packages wrap the `$PATH` binaries to provide `/nix/store` paths to applications.

**Comment by teto** [comment]
neovim now exposes plugins and config in $HOME in case you wanna switch to neovim

**Comment by genebean** [comment]
Thanks @teto!
